nengo/dists.py

```diff
@@ -350,11 +369,52 @@ class SubvectorLength(SqrtBeta):
     --------
     SqrtBeta
     """
+
     def __init__(self, dimensions, subdimensions=1):
         super(SubvectorLength, self).__init__(
             dimensions - subdimensions, subdimensions)
 
 
+class CosineSimilarity(SubvectorLength):
+    """Distribution of dot products between random unit vectors.
+
+    This is equivalent to the distribution of a single coefficient in a unit
+    vector (i.e. a single dimension of `UniformHypersphere(surface=True)`),
+    which can be obtained from `SubvectorLength` by symmetry.
+
+    This can be used to calculate an intercept `c = ppf(1 - p)` such that
+    `dot(u, v) >= c` with probability `p`, for random unit vectors `u` and `v`.
+    In other words, a neuron with intercept `ppf(1 - p)` will fire with
+    probability `p` for a random unit length input.
+
+    Parameters
+    ----------
+    dimensions: int
+        Dimensionality of the complete unit vector.
+
+    See also
+    --------
+    SqrtBeta
+    """
+
+    def __init__(self, dimensions):
+        super(CosineSimilarity, self).__init__(dimensions)
+
+    def sample(self, num, rng=np.random):
```

Comment:
`sample` doesn't have the same signature as we typically use for `Distribution.sample`. We might as well take the extra `d` parameter like `SubvectorLength` does, if it's not too much trouble (and I don't think it should be).